sdks/java/core/src/test/java/org/apache/beam/sdk/io/CompressedSourceTest.java

```diff
@@ -761,6 +1043,132 @@ public void testGzipProgress() throws IOException {
     }
   }
 
+  @Test
+  public void testEmptyLzoProgress() throws IOException {
+    File tmpFile = tmpFolder.newFile("empty.lzo_deflate");
+    String filename = tmpFile.toPath().toString();
+    writeFile(tmpFile, new byte[0], CompressionMode.LZO);
+
+    PipelineOptions options = PipelineOptionsFactory.create();
+    CompressedSource<Byte> source =
+        CompressedSource.from(new ByteSource(filename, 1)).withDecompression(CompressionMode.LZO);
+    try (BoundedReader<Byte> readerOrig = source.createReader(options)) {
+      assertThat(readerOrig, instanceOf(CompressedReader.class));
+      CompressedReader<Byte> reader = (CompressedReader<Byte>) readerOrig;
+      // before starting
+      assertEquals(0.0, reader.getFractionConsumed(), 1e-6);
```

Comment:
I see `1e-6` is used by many of the test cases as a threshold. Should it be factored out into a constant?

Reply:
It can be done. But that would require altering all the tests that use this constant value. Will that be fine if we do that?

Reply:
I think we can add the constant for `CompressedSourceTest.java` at least.

Reply:
we have added this in the update